### PR TITLE
dev: add tor from repo to standalone

### LINKED ISSUE #43: Integrate Tor in `standalone` container
~~Add the ability to compile the backend with `--with-local-tor` flag.~~
Tor is [a requirement as of v0.9.6](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/onion-message-channels.md#tor). Currently, the `standalone` image needs an external running Tor daemon to connect to.

The goal is to provide a truly independent "standalone" docker image, by adding a default Tor daemon to make use of it from the `jmwalletd` instance.

### PR DESCRIPTION
**Pull request by theborakompanioni**
Closes #43.

This PR adds Tor to the `standalone` image. 

Before this PR, an external container has been necessary to connect to IRC or to the directory nodes.
After this PR is applied, the image lives up to its name and is truly "standalone", meaning: No external dependencies. A Tor daemon will be started at the startup of the container.

The changes in this PR are up to debate and every input or suggestion is highly appreciated.

##### How to test?
This PR can be tested by following the instructions in [joinmarket-webui/#396 ](https://github.com/joinmarket-webui/joinmarket-webui/pull/396). A [build of this branch](https://github.com/joinmarket-webui/joinmarket-webui-docker/pkgs/container/joinmarket-webui-dev-standalone/28950781?tag=dev-with-local-tor-from-repo) has been made available as docker image, which is extended by the [secondary container in the regtest setup](https://github.com/joinmarket-webui/joinmarket-webui/blob/dev-regtest-with-local-tor/docker/regtest/dockerfile-deps/joinmarket/webui-standalone/Dockerfile#L1).

##### Addendum
Previous attempts included installing Tor from source via [flag `--with-local-tor` of joinmarket's `install.sh` script](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/v0.9.6/install.sh#L375-L398).
This has been working great locally, but led to longer build times on GitHub's environment (from ~40min to up to ~2h(!)) and subsequently failing at the unit tests for Tor ([[1]](https://github.com/joinmarket-webui/joinmarket-webui-docker/actions/runs/2692769830), [[2]](https://github.com/joinmarket-webui/joinmarket-webui-docker/actions/runs/2696178784), [[3]](https://github.com/joinmarket-webui/joinmarket-webui-docker/actions/runs/2696771104)). [Installing it from an official repository (via `apt-get`)](https://github.com/joinmarket-webui/joinmarket-webui-docker/blob/dev-with-local-tor-from-repo/standalone/Dockerfile#L75-L76) is a compromise between ease of use and robustness. Before, it was relied upon an external container, where the sources where also "unknown", or, as e.g. in Umbrels case, totally outside of the users control. It is always possible for a user to switch to another instance by adapting and passing configuration variables.

### CODE REVIEW

File: standalone/Dockerfile [hunk 1]
```diff
@@ -72,6 +72,8 @@ RUN apt-get update \
     tini supervisor iproute2 procps vim \
     # servers dependencies (see `install.sh`)
     build-essential automake pkg-config libtool libltdl-dev python3-dev python3-setuptools python3-pip \
+    # tor
+    tor \
```
Comment (dnlggr) [inline standalone/Dockerfile:76]: Not an expert but from what I can remember Tor recommends to [install from their own package repository](https://support.torproject.org/apt/tor-deb-repo/). Maybe this is something we should be doing here as well?

Comment (theborakompanioni) [reply standalone/Dockerfile:80]: Done. Great advise! Will do some testing and then report back once it can be reviewed again.

Comment (theborakompanioni) [reply standalone/Dockerfile:80]: Testing was successful. [Building the image now](https://github.com/joinmarket-webui/joinmarket-webui-docker/actions/runs/2703821256), will take ~1h and then it's ready to go!

**Review by dnlggr — APPROVED** [review]
Tested this as described in https://github.com/joinmarket-webui/joinmarket-webui/pull/396.

Both a scheduler round on instance A (the dev regtest build) with instance B (the standalone regtest build) as a maker and a single coinjoin on instance B with instance A as a maker worked.

I don't know anything about configuring Tor, though. So I can only acknowledge that everything worked on my machine (in regtest).

**Comment by theborakompanioni** [comment]
> I don't know anything about configuring Tor, though. So I can only acknowledge that everything worked on my machine (in regtest).

I will do some outstanding tests and merge after verifying that it works as expected.

**Comment by dergigi** [comment]
Same as dnlggr, I don't know much about this. Changeset looks good to me, though.

**Pull request merged**